LanguageFeatures/Class-modifiers/versioning_A03_t02.dart

```diff
@@ -0,0 +1,427 @@
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+abstract base mixin class AbstractBaseMixinClassImplementsPreFeatureExtendsFinal implements PreFeatureExtendsFinal {
+//                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+abstract base mixin class AbstractBaseMixinClassImplementsPreFeatureImplementsFinal implements PreFeatureImplementsFinal {
+//                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+abstract base mixin class AbstractBaseMixinClassImplementsPreFeatureWithFinal implements PreFeatureWithFinal {
+//                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+mixin MixinImplementsPreFeatureExtendsFinal implements PreFeatureExtendsFinal {
+//    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+mixin MixinImplementsPreFeatureImplementsFinal implements PreFeatureImplementsFinal {
+//    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+mixin MixinImplementsPreFeatureWithFinal implements PreFeatureWithFinal {
+//    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+base mixin BaseMixinImplementsPreFeatureExtendsFinal implements PreFeatureExtendsFinal {
+//         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+base mixin BaseMixinImplementsPreFeatureImplementsFinal implements PreFeatureImplementsFinal {
+//         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+base mixin BaseMixinImplementsPreFeatureWithFinal implements PreFeatureWithFinal {
+//         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+mixin MixinOnPreFeatureExtendsFinal on PreFeatureExtendsFinal {
+//    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+mixin MixinOnPreFeatureImplementsFinal on PreFeatureImplementsFinal {
+//    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+mixin MixinOnPreFeatureWithFinal on PreFeatureWithFinal {
+//    ^^^^^^^^^^^^^^^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  dynamic noSuchMethod(Invocation i) {}
+}
+
+base mixin BaseMixinOnPreFeatureExtendsFinal on PreFeatureExtendsFinal {
```

Comment:
These three cases are tricky, too.

We do have a rule that the `on` type of a mixin cannot be a `final` class of a different library, https://github.com/dart-lang/language/issues/2933, but I can't find that in the feature spec.

We have a similar rule for a `sealed` class in a different library ([here](https://github.com/dart-lang/language/blob/main/accepted/future-releases/class-modifiers/feature-specification.md#:~:text=A%20declaration%20depends%20directly%20on%20a%20sealed%20declaration%20from%20another%20library)).

But none of those rules are applicable here, so I'd expect this one plus the next two to be non-errors.

Reply:
Moved these cases to a separate test which is not expecting any errors

Reply:
As far as I can see we still have `MixinOnPreFeatureExtendsFinal`, `MixinOnPreFeatureImplementsFinal`, `MixinOnPreFeatureWithFinal` in LanguageFeatures/Class-modifiers/versioning_A03_t02.dart. Did you move some other declarations?

Reply:
No problem, I too missed that `on` relation is like an `implements` and restrictions for it are transitive.  Moved these tests to a separate test with appropriate description.